testplan/runnable/base.py

```diff
@@ -1072,6 +1072,11 @@ def make_runpath_dirs(self):
             )
             makedirs(self.resource_monitor_server_file_path)
 
+        if not os.environ.get("MPLCONFIGDIR"):
+            os.environ["MPLCONFIGDIR"] = os.path.join(
+                self._runpath, "matplotlib"
```

Comment:
does matplotlib complain if the dir pointed by MPLCONFIGDIR doesn't exist?

Reply:
No, it tries to create that.

This sets the path to the runpath, if the user didn't set the env var. This way there will be no warning, and the dir is only created when the matplotlib import executes. If I remove this default setting, there will be a warning when matplotlib is imported (and the default location is read-only).